src/go/k8s/pkg/resources/cluster-role.go

```diff
@@ -67,10 +66,7 @@ func (r *ClusterRoleResource) Ensure(ctx context.Context) error {
 	if errors.IsNotFound(err) {
 		r.logger.Info(fmt.Sprintf("ClusterRole %s does not exist, going to create one", r.Key().Name))
 
-		obj, err := r.Obj()
-		if err != nil {
-			return fmt.Errorf("unable to construct ClusterRole object: %w", err)
-		}
+		obj := r.obj()
```

Comment:
Why you don't use `resources.CreateIfNotExists(ctx, r, obj, r.logger)` here?

Reply:
good point, I don't know, you commited it :) we can do that, I would rather have that in another PR though